hops/featurestore.py

```diff
@@ -0,0 +1,2651 @@
+"""
+A feature store client. This module exposes an API for interacting with feature stores in Hopsworks.
+It hides complexity and provides utility methods such as:
+
+    - `project_featurestore()`.
+    - `get_featuregroup()`.
+    - `get_feature()`.
+    - `get_features()`.
+    - `sql()`
+    - `insert_into_featuregroup()`
+    - `get_featurestore_metadata()`
+    - `get_project_featurestores()`
+    - `get_featuregroups()`
+    - `get_training_datasets()`
```

Comment:
Will there be an API in future for something like: "get version X of the training dataset for featuregroup Y" ?

Reply:
That could be useful, I will look into it